server/src/modules/user/user.controller.ts

```diff
@@ -106,31 +118,62 @@ export class UserController {
     }
   }
 
-  @PublicRoute()
-  @SuperAdmin()
-  @Post('/admin')
-  async createNewAdminUser(
-    @Body() body: SignUpRequestBodyDto,
+  // get user information by id
+  @Get(':id')
+  async getUserById(
+    @Param() { id }: getUserDetailsByIdDto,
+  ): Promise<IUserResponse> {
+    try {
+      // call user service to extract user information
+      const user = await this.userService.getUserById(id);
+
+      return {
+        message: 'User information fetched successfully',
+        data: user,
+        status: HttpStatus.OK,
+      };
```

Comment:
You can create the common response-handler interceptor and send common response syntax for all the endpoints